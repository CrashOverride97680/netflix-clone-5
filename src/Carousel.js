import React from 'react';

const Carousel = props => <ul className="carousel">{props.children}</ul>;
