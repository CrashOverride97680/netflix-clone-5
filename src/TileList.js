import React, { Component } from 'react';
// import { Carousel } from 'react-responsive-carousel';
// import 'react-responsive-carousel/lib/styles/carousel.min.css';
import { discoverMovies } from './api/movieDB';
import Tile from './Tile';
import Carousel from './Carousel';

class TitleList extends Component {
  constructor(props) {
    super(props);
    this.state = { movies: [] };
    this.startCarousel = -1;
  }

  componentWillMount() {
    discoverMovies()
      .then((response) => {
        const data = response.data.results;
        const movies = this.renderMovieTiles(data);
        this.setState({ movies });
      })
      .catch((error) => {
        console.log(error);
      });
  }

  shiftRight() {
    const movies = this.state.movies;
    let start = 1;
    for (let i = this.startCarousel + 1; i < movies.length; ++i) {
      movies[i] = React.cloneElement(movies[i], { style: { order: start++ } });
    }
    for (let i = this.startCarousel; i >= 0; --i) {
      movies[i] = React.cloneElement(movies[i], { style: { order: movies.length } });
    }
    this.setState({ movies });
  }

  renderMovieTiles(data) {
    let movies = [];
    if (data.length > 0) {
      const order = new Array(data.length);
      for (let i = 0; i < data.length; ++i) {
        order[i] = (i + 1) % data.length;
      }
      let i = -1;
      movies = data.map((movie) => {
        i += 1;
        return (
          <Tile
            key={movie.id}
            imgUrl={movie.backdrop_path}
            title={movie.title}
            style={{ order: order[i] }}
          />
        );
      });
      /*
      React.cloneElement(content[content.length - 1], {
        className: 'is-ref',
      });

      movies[movies.length - 1].props.style = { order: '1' };
      */
      this.startCarousel = movies.length - 1;
    }
    return movies;
  }

  render() {
    if (this.state.movies.length > 0) {
      return (
        <div>
          <button>Shift right</button>
          <Carousel>{this.state.movies}</Carousel>
        </div>
      );
    }
    return <div />;
  }
}
export default TitleList;
