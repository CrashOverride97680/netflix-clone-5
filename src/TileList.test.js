import React from 'react';
import { shallow, mount } from 'enzyme';
import { createWaitForElement } from 'enzyme-wait';

import TileList from './TileList';
import Carousel from './Carousel';

describe('TileList', () => {
  it('Compiles', () => {
    jest.mock('axios', () => ({ get: jest.fn() }));
    const wrapper = shallow(<TileList />);
    expect(wrapper.find('div').length).toBe(1);
  });
  it('Displays movies once the api resolves', () => {
    // axios.get = jest.fn(url => get(url));
    jest.mock('./api/__mocks__/axios');
    const waitForSample = createWaitForElement(Carousel);
    expect.assertions(2);
    const wrapper = mount(<TileList />);
    return waitForSample(wrapper).then((wrapper) => {
      const movies = wrapper.find(Carousel).html();
      expect(wrapper.state().movies.length).toBe(20);
      return expect(movies).toMatchSnapshot();
    });
  });
});
