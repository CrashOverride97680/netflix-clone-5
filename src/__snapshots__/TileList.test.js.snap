// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TileList Displays movies once the api resolves 1`] = `"<div class=\\"wrap\\"><ul class=\\"carousel\\"><div class=\\"item\\"><img src=\\"https://image.tmdb.org/t/p/original/uX7LXnsC7bZJZjn048UCOwkPXWJ.jpg\\" alt=\\"Movie Minions backdrop\\"></div><div class=\\"item\\"><img src=\\"https://image.tmdb.org/t/p/original/tcheoA2nPATCm2vvXw2hVQoaEFD.jpg\\" alt=\\"Movie It backdrop\\"></div><div class=\\"item\\"><img src=\\"https://image.tmdb.org/t/p/original/6aUWe0GSl69wMTSWWexsorMIvwU.jpg\\" alt=\\"Movie Beauty and the Beast backdrop\\"></div><div class=\\"item\\"><img src=\\"https://image.tmdb.org/t/p/original/6iUNJZymJBMXXriQyFZfLAKnjO6.jpg\\" alt=\\"Movie Wonder Woman backdrop\\"></div><div class=\\"item\\"><img src=\\"https://image.tmdb.org/t/p/original/5pAGnkFYSsFJ99ZxDIYnhQbQFXs.jpg\\" alt=\\"Movie Logan backdrop\\"></div><div class=\\"item\\"><img src=\\"https://image.tmdb.org/t/p/original/aJn9XeesqsrSLKcHfHP4u5985hn.jpg\\" alt=\\"Movie Guardians of the Galaxy Vol. 2 backdrop\\"></div><div class=\\"item\\"><img src=\\"https://image.tmdb.org/t/p/original/vc8bCGjdVp0UbMNLzHnHSLRbBWQ.jpg\\" alt=\\"Movie Spider-Man: Homecoming backdrop\\"></div><div class=\\"item\\"><img src=\\"https://image.tmdb.org/t/p/original/ftRkFtAGuHngHnLiOxktq0aCVMF.jpg\\" alt=\\"Movie Despicable Me 3 backdrop\\"></div><div class=\\"item\\"><img src=\\"https://image.tmdb.org/t/p/original/tZvdyLP2F6x2TTuh292ceH87qZT.jpg\\" alt=\\"Movie Pirates of the Caribbean: Dead Men Tell No Tales backdrop\\"></div><div class=\\"item\\"><img src=\\"https://image.tmdb.org/t/p/original/7KsqfXDECZMryX1Rv4RKsT7SIjQ.jpg\\" alt=\\"Movie The Hitman's Bodyguard backdrop\\"></div><div class=\\"item\\"><img src=\\"https://image.tmdb.org/t/p/original/kMU8trT43p5LFoJ4plIySMOsZ1T.jpg\\" alt=\\"Movie Alien: Covenant backdrop\\"></div><div class=\\"item\\"><img src=\\"https://image.tmdb.org/t/p/original/t6TL7V9IsT4Ktx6YO8b1UYj3agd.jpg\\" alt=\\"Movie Baywatch backdrop\\"></div><div class=\\"item\\"><img src=\\"https://image.tmdb.org/t/p/original/goCvLSUFz0p7k8R10Hv4CVh3EQv.jpg\\" alt=\\"Movie Baby Driver backdrop\\"></div><div class=\\"item\\"><img src=\\"https://image.tmdb.org/t/p/original/tFI8VLMgSTTU38i8TIsklfqS9Nl.jpg\\" alt=\\"Movie Doctor Strange backdrop\\"></div><div class=\\"item\\"><img src=\\"https://image.tmdb.org/t/p/original/fudEG1VUWuOqleXv6NwCExK0VLy.jpg\\" alt=\\"Movie Dunkirk backdrop\\"></div><div class=\\"item\\"><img src=\\"https://image.tmdb.org/t/p/original/6I2tPx6KIiBB4TWFiWwNUzrbxUn.jpg\\" alt=\\"Movie Fantastic Beasts and Where to Find Them backdrop\\"></div><div class=\\"item\\"><img src=\\"https://image.tmdb.org/t/p/original/yIZ1xendyqKvY3FGeeUYUd5X9Mm.jpg\\" alt=\\"Movie Arrival backdrop\\"></div><div class=\\"item\\"><img src=\\"https://image.tmdb.org/t/p/original/o8u0NyEigCEaZHBdCYTRfXR8U4i.jpg\\" alt=\\"Movie Annabelle: Creation backdrop\\"></div><div class=\\"item\\"><img src=\\"https://image.tmdb.org/t/p/original/4G6FNNLSIVrwSRZyFs91hQ3lZtD.jpg\\" alt=\\"Movie Split backdrop\\"></div><div class=\\"item\\"><img src=\\"https://image.tmdb.org/t/p/original/tZjVVIYXACV4IIIhXeIM59ytqwS.jpg\\" alt=\\"Movie Rogue One: A Star Wars Story backdrop\\"></div></ul></div>"`;
